integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/DeviceState.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.cucumber.platform.publiclighting.mocks.oslpdevice;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+
+import org.opensmartgridplatform.cucumber.core.ScenarioContext;
+import org.opensmartgridplatform.cucumber.platform.PlatformKeys;
+import org.opensmartgridplatform.oslp.Oslp;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+
+import lombok.Getter;
+import lombok.Setter;
+
+public class DeviceState {
+
+    @Getter
+    private final String deviceUID;
+    @Getter @Setter
+    private Integer sequenceNumber = 0;
```

Comment:
Good point, but in this case, I don't think it really is an issue. There is only one class from where the sequencenumber is retrieved or changed, and that is from MockOslpChannelHandler. There is only one instance of that class, and that's not shared over threads.

The fact that ConcurrentMaps are used, is because there is a mock server thread, and a 'cucumber' thread. The cucumber thread adds mocks to the map, and the mock server removes them.